xgi/readwrite/xgi_data.py

```diff
@@ -32,15 +39,94 @@ def load_xgi_data(dataset, nodetype=None, edgetype=None, max_order=None):
     XGIError
        The specified dataset does not exist.
     """
-    index_url = "https://raw.githubusercontent.com/ComplexGroupInteractions/xgi-data/main/index.json"
+
+    if read:
+        cfp = _make_unix_file_path(path, dataset)
+        if os.path.exists(cfp):
+            data = json.load(open(cfp, 'r'))
+        else:
+            data = _request_from_xgi_data(dataset)
+    else:
+        data = _request_from_xgi_data(dataset)
+
+    return convert.dict_to_hypergraph(
+        data, nodetype=nodetype, edgetype=edgetype, max_order=max_order
+    )
+
+
+def download_xgi_data(dataset, path=''):
+    """Make a local copy of a dataset in the xgi-data repository.
+
+    Parameters
+    ----------
+    dataset : str
+        Dataset name. Valid options are the top-level tags of the
+        index.json file in the xgi-data repository.
+
+    path : str, optional
+        Path to where the local copy should be saved. If none is given, save
+        file to local directory.
+    """
+
+    jsondata = _request_from_xgi_data(dataset)
+    jsonfile = open(_make_unix_file_path(path, dataset), 'w')
+    json.dump(jsondata, jsonfile)
+    jsonfile.close()
+
+
+def _make_unix_file_path(path, dataset):
+    """Create a path to file in unix format to an xgi data set in a jason file.
+    
+    Parameters
+    ----------
+    path : str
+        Path to a local directory.
+
+    dataset : str
+        Dataset name. Valid options are the top-level tags of the
+        index.json file in the xgi-data repository.
+
+    Returns
+    -------
+    Filepath
+        The unix-format file path.
+    """
+
+    if len(path)>0:
+        if path[-1] not in ['/', '\\']:
+            path = path + '/'
+        elif path[-1]=='\\':
+            path = path.replace('\\', '/')
+
+    filepath = path + dataset + '.json'
+
+    return filepath  
+
+
+def _request_from_xgi_data(dataset):
+    """Request a dataset from xgi-data.
+
+    Parameters
+    ----------
+    dataset : str
+        Dataset name. Valid options are the top-level tags of the
+        index.json file in the xgi-data repository.
+
+    Returns
+    -------
+    Data
+        The requested data loaded from a json file.
```

Comment:
```suggestion
        The requested data loaded from a json file.
        
   See also
   ---------
   load_xgi_data
```

Could you also add a See also to load_xgi_data with download_xgi_data?